File: core/src/test/java/de/yard/threed/core/testutil/TestFactory.java
package de.yard.threed.core.testutil;

import de.yard.threed.core.InitMethod;
import de.yard.threed.core.Util;
import de.yard.threed.core.configuration.Configuration;
import de.yard.threed.core.configuration.ConfigurationByArgs;
import de.yard.threed.core.configuration.ConfigurationByEnv;
import de.yard.threed.core.configuration.ConfigurationByProperties;
import de.yard.threed.core.platform.Platform;
import de.yard.threed.core.platform.PlatformFactory;
import de.yard.threed.core.platform.PlatformInternals;

import java.util.HashMap;

public class TestFactory {

    public static Platform initPlatformForTest(PlatformFactory platformFactory, InitMethod sceneIinitMethod) {

        // Some projects tests need env variables like "ADDITIONALBUNDLE"
        HashMap<String, String> properties = new HashMap<String, String>();
        PlatformInternals pl = platformFactory.createPlatform(Configuration.buildDefaultConfigurationWithEnv(properties)
                .addConfiguration(new ConfigurationByEnv(), false));

        return Platform.getInstance();
    }

}

File: platform-unity/PlatformUnity/Assets/scripts-platform/TestFactory.cs
using System;
using java.lang;
namespace de.yard.threed.platform {

using de.yard.threed.engine;
using de.yard.threed.core.platform;
using de.yard.threed.core;
using de.yard.threed.platform.unity;
    using de.yard.threed.core.configuration;

    /**
     *
     */
    public class TestFactory  {
    //Log logger = Platform.getInstance().getLog(typeof(UnityBase3D));
    
        public static Platform initPlatformForTest(){
            PlatformInternals pli = PlatformUnity.init(new ConfigurationByProperties(new java.util.HashMap<string, string>()));
            PlatformUnity pl = (PlatformUnity)Platform.getInstance ();
            pl.logfactory = new TestLogFactory();
            return pl;
        }

   
}
}